codalab/server/authenticated_plugin.py

```diff
@@ -1,52 +1,87 @@
+from abc import ABC, abstractmethod
 from bottle import abort, httplib, redirect, request, url
 
 from codalab.lib.server_util import redirect_with_query
 from codalab.objects.user import PUBLIC_USER
 
+import os
 
-def user_is_authenticated():
-    return hasattr(request, 'user') and request.user is not None and request.user is not PUBLIC_USER
 
+class AuthenticationPlugin(ABC):
```

Comment:
Can we rename this to `AuthPlugin`? This will make it clearer and less easy to confuse with `AuthenticatedPlugin`.